src/main/java/com/powsybl/caseserver/datasource/util/S3CaseDataSourceService.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.caseserver.datasource.util;
+
+import com.powsybl.caseserver.server.CaseService;
+import com.powsybl.caseserver.server.S3CaseService;
+import com.powsybl.commons.datasource.DataSource;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.Functions.FailableFunction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author Ghazwa Rehili <ghazwa.rehili at rte-france.com>
+ */
+@Service
+@ComponentScan(basePackageClasses = CaseService.class)
+public class S3CaseDataSourceService implements CaseDataSourceService {
+
+    @Autowired
+    private S3CaseService caseService;
+
+    @Override
+    public String getBaseName(UUID caseUuid) {
+        return withS3DownloadedDataSource(caseUuid, DataSource::getBaseName);
```

Comment:
should not download the whole file just to get the basename

Reply:
done